src/Camera.py

```diff
@@ -6,31 +6,37 @@
 class Camera:
     """Abstraction of the cv2 VideoCapture device."""
 
-    TARGET_RESOLUTION_HEIGHT = 480
+    COMMON_RESOLUTIONS = [(640, 480), (854, 480), (1280, 720), (1920, 1080)]
 
     def __init__(self) -> None:
         print("Initializing camera...")
         # Get access to the VideoCapture device
         self.capture = cv2.VideoCapture(0)
-        # Calculate an adjusted resolution
-        baseWidth = self.capture.get(cv2.CAP_PROP_FRAME_WIDTH)
-        baseHeight = self.capture.get(cv2.CAP_PROP_FRAME_HEIGHT)
-
-        print(f"Camera resolution: {baseWidth}x{baseHeight}")
-
-        factor = float(Camera.TARGET_RESOLUTION_HEIGHT / baseHeight)
-
-        resolutionWidth = int(baseWidth * factor)
-
-        self.resolution: tuple[int] = (resolutionWidth, Camera.TARGET_RESOLUTION_HEIGHT)
-
-        print(
-            f"Adjusted resolution: {resolutionWidth}x{Camera.TARGET_RESOLUTION_HEIGHT}"
-        )
+        # Get the minimum supported common resolution
+        actualW = self.capture.get(cv2.CAP_PROP_FRAME_WIDTH)
+        actualH = self.capture.get(cv2.CAP_PROP_FRAME_HEIGHT)
+        print(f"Actual camera resolution: {actualW}x{actualH}")
+        self.resolution: tuple[int] = min(self.getSupportedResolutions())
+        print(f"Adjusted camera resolution: {self.resolution[0]}x{self.resolution[1]}")
         # Adjust the resolution
         self.capture.set(cv2.CAP_PROP_FRAME_WIDTH, self.resolution[0])
         self.capture.set(cv2.CAP_PROP_FRAME_HEIGHT, self.resolution[1])
 
+    def getSupportedResolutions(self) -> list[tuple[int]]:
+        resolutions = []
+
+        for res in Camera.COMMON_RESOLUTIONS:
+            # Set the camera frame to that size
+            self.capture.set(cv2.CAP_PROP_FRAME_WIDTH, res[0])
+            self.capture.set(cv2.CAP_PROP_FRAME_HEIGHT, res[1])
```

Comment:
What's the point of setting the camera resolution many times, only to return a list of values, and then set it again?